### DISCUSSION TITLE
How can I use the ROM import function correctly?

### DISCUSSION THREAD

**Discussion by VerdichteteEnergie** [hneemann/Digital#1161] [General]
I use a ROM for my small programs, which I compile into binary form and use for my computer based on the nand2tetris courses. For this I write the programs directly into the ROM by opening the ROM chip in digital and directly writing the binary codes into each address via edit.

Now before I want to write longer programs, I tried to add the programs in pure 16-bit binary words to digital via the ROM chip's import function. Unfortunately, these are not accepted, because when I click on edit, no code can be seen in the ROM. And if I try to start the program, an error pop up tells me, that my ROM is not in intel hex format.

I saved my program with simple 16-bit words per line in a file called "rom-test.hex".

Here is my example code:
```
0000000000001101
1111110000010000
0100000000000000
1110000010100000
1110111010001000
0000000000001101
1111110000010000
1110011111010000
1110001100001000
0000000000000000
1110101010000111
```

Here my file:
[rom-test.zip](https://github.com/hneemann/Digital/files/11544942/rom-test.zip)

I've read through the official documentation, but unfortunately couldn't find anything about it.

Can someone help me and show what I'm doing wrong?

Operating system:
Windows 11 (64-Bit) Pro

**Comment by hneemann** [comment]
The import supports the old Logisim hex file format, Intel hex files or simple plain binary files.
The old Logisim Hex format is the easiest to use:
```
v2.0 raw
0a0f
10e1
2be0
10e1
0af7

```
The first line identifies the format, and then the data words follow in hexadecimal format. The file extension needs to be *.hex.

**Reply by VerdichteteEnergie** [reply]
Thank you again!

It works perfectly!
